### PR TITLE
feat: badger day update

### LINKED ISSUE #13: Allow Owners to Archive Badges/Organizations
Allow Owners to hide Badges and Organizations from the front end.

This does not impact anything on-chain.

The Owner may "Archive" either the Badger or the Organization to hide them from their view.

### LINKED ISSUE #116: Fix: Being Badgered To Reconnect When Using Another Tab
Bug report:

> one thing I've noticed is that if you go to another chrome tab and do something on another network, badger keeps badgering you to switch back to polygon and resign with badger

While the website is functioning as designed, we should consider this. I want to find a solution because I also experienced this while bridging funds into my Organization the other night, and it was less than ideal.

### LINKED ISSUE #133: Fix: Front-End API Integration
Right now, the API integration on the front end is setup incorrectly, resulting in stale data overwriting newer data. Additionally, due to the existing setup, if an organization has 5,000 members, it will expect all 5,000 members to be posted.

Most importantly, though, due to this current setup, balances cannot synchronously be tracked.

### LINKED ISSUE #134: Feature: Synchronous Balance Tracking
Dependent on #133, that will enable the ability for improved API integration and the more granular ability to have unique interaction and lower transfer of data.

### LINKED ISSUE #137: feat: organization transfer
Implement the API and front-end support for the danger zone of organization drawer design.

### LINKED ISSUE #156: Fix: Transfered Orgs Not Visible for New Owner
Transferred org (PFP Deployer) from danner.eth to directdeposit.eth

https://polygonscan.com/tx/0x42caa0ac4a359ab14580c212cb5fbc09e00a2b3165fae93690b61216adb6e223

org: 0x90ffdF48A1660A2759d82c1aF78318F414B1a9b5
original owner: 0x581bef12967f06f2ebfcabb7504fa61f0326cd9a
new owner: 0x34744Dd0Fb927366228A2eAee0D142620FF26077

view from new owner

image

view from old owner 

image

(note: before logging out and back in to danner.eth, I could still see the PFP Deployer org and I could only see the item in it that danner.eth was a holder of, but not a manager of)

### LINKED ISSUE #157: Bug: Can't switch wallets
Right now when switching wallets it pings back:

> Account was changed -- Please sign in or switch back to 0xf39F...2266.

For some reason? When a wallet is changed, it should change.

### LINKED ISSUE #173: Fix: Clear address form after tx confirmation

### PR DESCRIPTION
**Pull request by 381e35**
PR #174 rebuilds most of the foundation of the web-app to operate on the framework needed to leave the MVP phase. With this, WebSockets become integrated, followed by clean-up in the supporting areas.

We have had to handle a bit of technical debt in the beginning; thus, this PR rebuilds every piece of the app. While we knew that we would one day want WebSockets, we did not expect to change our approach in such a severe manner.

To support this, the pieces are being developed in parallel to make the:
- API
- Frontend
- Smart Contract

All work together without compromise. Avoiding the typical downfall of traditional web3 app development, the on-chain team is the same as the off-chain; thus, we can secure benefits built into the product's core.

Moving forward, the technical overhead and developer onboarding are significantly lower due to the approachability and general architecture of the API and the consuming routes.

Once the app experience has been updated, the brand will also be updated in this same PR.

- [x] Resolved by #207 with the deprecation of auth: Authentication is nice and smooth.
- [x] #197
- [x] #198
- [x] c2b041c40e1d9e096ffd7c232213f94fc4dbf043 : Implement API support for balances and Badges as source.
- [x] #211 Frontend consumption of #198 & 658c7f4b363a5aa93c720aae2d5a574272856f30.
- [x] #206 Use the Badge websocket source as the data source for Badges.
     - [x] This will require slight data shifting up one level to drill down.
- [x] Badge managers (just people no configuration) on the frontend.
     
**Nice to haves:**

- [x] bba074633e8fe6f6d95bd114d42d98aa9dc50d12 : Refactor background jobs to be balancing / non-critical functionality.
    - [x] Make sure that we are constantly updated with balances.
    - [x] Refresh the static metadata of the Organization every so often.
    - [x] Fill in IPFS metadata asynchronously instead of blocking.

### CODE REVIEW

File: frontend/src/pages/Badge/BadgeForm.js [hunk 1]
```diff
@@ -17,320 +18,120 @@ import {
...
-
-    const setBadge = useSetBadge(
-        !isDisabled,
-        orgData.ethereum_address,
-        badgeImage ? deterministicMetadataHash : badgeData?.token_uri,
-        badge
-    );
+    const { authenticatedAddress, chain, organization, badge } = useUser({ orgId, badgeId });
+
+    const [ obj, dispatchObj ] = useReducer(FormReducer, badge || initialBadgeForm);
```
Comment (nftchance) [inline frontend/src/pages/Badge/BadgeForm.js:31]: I need a crash course on this.

File: frontend/src/hooks/useLogout.js [hunk 2]
```diff
@@ -11,10 +11,12 @@ const useLogout = () => {
     const logout = () => {
         // TODO: Reenable this once the websocket is prevented from trying to reconnect when not enabled
         // document.cookie = 'csrftoken=; Path=/; Expires=Sat, 01 Jan 2000 00:00:001 GMT;';
-
+        
+        // TODO: We cannot clear the sessionid cookie because it is an HTTP_ONLY cookie, so we need to find
```
Comment (nftchance) [inline frontend/src/hooks/useLogout.js:15]: The answer is not to remove things. Yes, I am aware :)

Comment (nftchance) [reply frontend/src/hooks/useLogout.js:15]: The above commented-out line has the answer already ^

File: frontend/src/contexts/AuthenticationContext.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ const AuthenticationContextProvider = ({ children }) => {
     const isAuthenticated = isConnected && !isWrongNetwork && address === authenticatedAddress;
 
     useEffect(() => {
-        if (isWrongNetwork) switchNetwork(primaryChain.id)
+        if (isWrongNetwork) switchNetwork?.(primaryChain.id)
```
Comment (nftchance) [inline frontend/src/contexts/AuthenticationContext.js:28]: Proper way of doing this is:

`if (isAuthenticated && isWrongNetwork)`

Comment (masonchain) [reply frontend/src/contexts/AuthenticationContext.js:28]: Tried that, still was crashing with "switchNetwork is not a function"

Comment (nftchance) [reply frontend/src/contexts/AuthenticationContext.js:28]: Yeah I mean I'm no Wagmi expert I don't know the actual library, I just meant please let's ignore the awful syntax and approach that they suggest; it is very bad

**Comment by nftchance** [comment]
- [ ] Indexer running locally, so we can use that for the backfill data source.
- [ ] Authentication is nice and smooth.
- [ ] Need to get all the forms connected again.
- [ ] Finalize the indexer to index Managers and Hooks.
- [ ] Use the Badge websocket source as the data source for Badges.
     - [ ] This will require slight data shifting up one level to drill down.

**Pull request closed without merging**